baselines/NER/my_roberta_multilabel.py

```diff
@@ -0,0 +1,413 @@
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        outputs = self.roberta(
+            input_ids,
+            attention_mask=attention_mask,
+            token_type_ids=token_type_ids,
+            position_ids=position_ids,
+            head_mask=head_mask,
+            inputs_embeds=inputs_embeds,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        if self.use_1d_positional_embeddings:
+            input_shape = input_ids.size()
+            seq_length = input_shape[1]
+            if position_ids is None:
+                position_ids = torch.arange(seq_length, dtype=torch.long, device=input_ids.device)
+                position_ids = position_ids.unsqueeze(0).expand_as(input_ids)
+            position_emb = self.pos_emb(position_ids)
+
+        if self.use_2d_positional_embeddings:
+            # create embeddings (each coordinate separate)
+            # bb_left_emb = self.bb_left_emb(bboxes[:, 0])
+            bb_left_emb = self.bb_left_emb(bboxes[:, :, 0])
+            # bb_top_emb = self.bb_top_emb(bboxes[:, 1])
+            bb_top_emb = self.bb_top_emb(bboxes[:, :, 1])
+            # bb_right_emb = self.bb_right_emb(bboxes[:, 2])
+            bb_right_emb = self.bb_right_emb(bboxes[:, :, 2])
+            # bb_bottom_emb = self.bb_bottom_emb(bboxes[:, 3])
+            bb_bottom_emb = self.bb_bottom_emb(bboxes[:, :, 3])
+            # n_rep = outputs[0].shape[1]
+
+            # final bbox embedding is a sum of all coordinate embeddings
+            # bbox_embedding = bb_top_emb.unsqueeze(1).expand(-1, n_rep, -1) + bb_left_emb.unsqueeze(1).expand(-1, n_rep, -1) + bb_bottom_emb.unsqueeze(1).expand(-1, n_rep, -1) + bb_right_emb.unsqueeze(1).expand(-1, n_rep, -1)
+            # bbox_embedding = bb_top_emb + bb_left_emb + bb_bottom_emb + bb_right_emb
+            if self.use_2d_concat:
+                bbox_embedding = torch.cat(
+                    [bb_top_emb, bb_left_emb, bb_bottom_emb, bb_right_emb], dim=-1
+                )
+            else:
+                bbox_embedding = bb_top_emb + bb_left_emb + bb_bottom_emb + bb_right_emb
+
+        if self.use_new_2D_pos_emb:
+            l = bboxes[:, :, 0]  # noqa: E741
+            t = bboxes[:, :, 1]
+            r = bboxes[:, :, 2]
+            b = bboxes[:, :, 3]
+            cx = (l + r) / 2
+            cy = (t + b) / 2
+            w = r - l
+            h = b - t
+            pos2_cx_emb = self.pos2_cx_emb((cx / self.quant_step_size + 0.5).int())
+            pos2_cy_emb = self.pos2_cx_emb((cy / self.quant_step_size + 0.5).int())
+            pos2_w_emb = self.pos2_cx_emb((w / self.quant_step_size + 0.5).int())
+            pos2_h_emb = self.pos2_cx_emb((h / self.quant_step_size + 0.5).int())
```

Comment:
There is a bug in here, `pos2_cx_emb` is used everywhere instead of using the correct `pos2_X_emb`, respectively